src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -479,6 +479,26 @@ private IOperation CreateBoundUnconvertedAddressOfOperatorOperation(BoundUnconve
                 boundUnconvertedAddressOf.WasCompilerGenerated);
         }
 
+        private IOperation CreateBoundAttributeOperation(BoundAttribute boundAttribute)
+        {
+            var isAttributeImplicit = boundAttribute.WasCompilerGenerated;
+            if (boundAttribute.HasErrors)
```

Comment:
We might want to imitate what we do for e.g. object creation, rather than always creating an "invalid" operation when the bound node has errors:

https://github.com/dotnet/roslyn/blob/b39705873a2ef5771c2d797246bcbdcac3d86fbc/src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs#L660

This might mean adding a ResultKind field to BoundAttribute.

Reply:
This won't produce invalid operation for errors that are specific to attributes, for example, the `ERR_BadAttributeParamType` for which we moved the `ValidateTypeForAttributeParameters` call. If we're okay with not producing `IInvalidOperation` for this, I'll revert the `ValidateTypeForAttributeParameters` change.

Reply:
> This won't produce invalid operation for errors that are specific to attributes, for example, the `ERR_BadAttributeParamType` for which we moved the `ValidateTypeForAttributeParameters` call. If we're okay with not producing `IInvalidOperation` for this, I'll revert the `ValidateTypeForAttributeParameters` change.

@333fred for thoughts on what should generate `IInvalidOperation` vs. what shouldn't.

Reply:
And invalid operation should be generated when we have no idea what information to give to a consumer. I'd be fine with us giving more info where possible, but I think that should be a separate PR with its own set of tests. We should probably leave this as is for now.

Reply:
The current approach requires me to update the binder for some cases to have `HasError` to true for some cases where the errors used to be reported after the node is bound.

I think it's better to revert those binding changes, and only produce `IInvalidOperation` when we have no information.